lib/roger/resolver.rb

```diff
@@ -176,7 +176,7 @@ def sanitize_name(name, prefer = nil)
       path = name.to_s
 
       # If it's a directory append "index"
-      path = File.join(path, "index") if File.directory?(name)
+      path = File.join(path, "index") if File.directory?(name) && Dir.glob(name + ".*").none?
```

Comment:
Well it adds index anyway, so it will never resolve to bla.html.erb.

Very edge cases, but I would like to be able to:

```
- elements
  - footer/
     - disclaimer
  - footer.html.erb
```

```
  <%= partial 'elements/footer' %>
```

Right now it just blows up, because it appends index to the name on: https://github.com/DigitPaint/roger/blob/fix-dir-and-partial-with-same-name/lib/roger/resolver.rb#L179